ui/app/components/secret-delete-menu.js

```diff
@@ -24,92 +22,21 @@ export default class SecretDeleteMenu extends Component {
 
   @tracked showDeleteModal = false;
 
-  @maybeQueryRecord(
-    'capabilities',
-    (context) => {
-      if (!context.args || !context.args.modelForData || !context.args.modelForData.id) return;
-      const [backend, id] = JSON.parse(context.args.modelForData.id);
-      return {
-        id: `${backend}/undelete/${id}`,
-      };
-    },
-    'model.id'
-  )
-  undeleteVersionPath;
-  @alias('undeleteVersionPath.canUpdate') canUndeleteVersion;
-
-  @maybeQueryRecord(
-    'capabilities',
-    (context) => {
-      if (!context.args || !context.args.modelForData || !context.args.modelForData.id) return;
-      const [backend, id] = JSON.parse(context.args.modelForData.id);
-      return {
-        id: `${backend}/destroy/${id}`,
-      };
-    },
-    'model.id'
-  )
-  destroyVersionPath;
-  @alias('destroyVersionPath.canUpdate') canDestroyVersion;
-
-  @maybeQueryRecord(
-    'capabilities',
-    (context) => {
-      if (!context.args.model || !context.args.model.engine || !context.args.model.id) return;
-      const backend = context.args.model.engine.id;
-      const id = context.args.model.id;
-      return {
-        id: `${backend}/metadata/${id}`,
-      };
-    },
-    'model',
-    'model.id',
-    'mode'
-  )
-  v2UpdatePath;
-  @alias('v2UpdatePath.canDelete') canDestroyAllVersions;
-
-  @maybeQueryRecord(
-    'capabilities',
-    (context) => {
-      if (!context.args.model || context.args.mode === 'create') {
-        return;
-      }
-      const backend = context.args.isV2 ? context.args.model.engine.id : context.args.model.backend;
-      const id = context.args.model.id;
-      const path = context.args.isV2 ? `${backend}/data/${id}` : `${backend}/${id}`;
-      return {
-        id: path,
-      };
-    },
-    'isV2',
-    'model',
-    'model.id',
-    'mode'
-  )
-  secretDataPath;
-  @alias('secretDataPath.canDelete') canDeleteSecretData;
-
-  @maybeQueryRecord(
-    'capabilities',
-    (context) => {
-      if (!context.args.model || context.args.mode === 'create') {
-        return;
-      }
-      const backend = context.args.isV2 ? context.args.model.engine.id : context.args.model.backend;
-      const id = context.args.model.id;
-      const path = context.args.isV2 ? `${backend}/delete/${id}` : `${backend}/${id}`;
-      return {
-        id: path,
-      };
-    },
-    'isV2',
-    'model',
-    'model.id',
-    'mode'
-  )
-  secretSoftDataPath;
-  @alias('secretSoftDataPath.canUpdate') canSoftDeleteSecretData;
+  get canUndeleteVersion() {
```

Comment:
This component should actually go away soon, so to simplify the changeset I just made these all getters of the same name